src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+|===
+
+=== Indirect Access S-mode CSRs
+
+If an interrupt _i_ is not present in the hardware, the corresponding CLIC register
+locations appear hardwired to zero.
+
+In S-mode, if an interrupt _i_ is not accessible to S-mode, the corresponding CLIC register
+locations appear hardwired to zero.
+
+==== `clicintattr[__i__]`
+
+In this `siselect` offset range:
+
+* Each `sireg` register controls the clic level/priority setting of four interrupts
+* Each `sireg2` register controls the clic attribute setting of four interrupts
+
+[%autowidth]
+|===
+| `siselect`   |  `sireg2` bits |  `sireg2` state          | description
+
+| 0x1000+_i_   |   7:0        | RW  `clicintattr[_i_*4+0]` |  setting for interrupt _i_*4+0
+| 0x1000+_i_   |  15:8        | RW  `clicintattr[_i_*4+1]` |  setting for interrupt _i_*4+1
+| 0x1000+_i_   |  23:16       | RW  `clicintattr[_i_*4+2]` |  setting for interrupt _i_*4+2
+| 0x1000+_i_   |  31:24       | RW  `clicintattr[_i_*4+3]` |  setting for interrupt _i_*4+3
+|===
+
+==== `cliciprio[__i__]`
+
+In this `siselect` offset range:
+
+* Each `sireg3` register controls the clic attribute setting of four interrupts
+
+[%autowidth]
+|===
+| `siselect` |  `sireg3` bits |  `sireg3` state              | description
+
+| 0x1000+i   |   7:0          | RW  `cliciprio[__i__*4+0]` |  setting for interrupt __i__*4+0
+| 0x1000+i   |  15:8          | RW  `cliciprio[__i__*4+1]` |  setting for interrupt __i__*4+1
+| 0x1000+i   |  23:16         | RW  `cliciprio[__i__*4+2]` |  setting for interrupt __i__*4+2
+| 0x1000+i   |  31:24         | RW  `cliciprio[__i__*4+3]` |  setting for interrupt __i__*4+3
+|===
+
+
+==== `clicintip[__i__]` and `clicintie[__i__]`
+
+In this `siselect` offset range:
+
+* Each `sireg` register controls the interrupt pending of thirty-two interrupts.
+* Each `sireg2` register controls the interrupt enable of thirty-two interrupts.
+
+[%autowidth]
+|===
+| `siselect`   |  `sireg` bits |  `sireg` state          |  `sireg2` bits |  `sireg2` state          | description
+
+| 0x1400    |   31:0   | RW `clicintip[31:0]`      |   31:0    | RW `clicintie[31:0]`       | settings for interrupts 31 through 0
+| 0x1401    |   31:0   | RW `clicintip[63:32]`     |   31:0    | RW `clicintie[63:32]`      | settings for interrupts 63 through 32
+6*^| ...
+| 0x147F    |   31:0   | RW `clicintip[4095:4064]` |   31:0    | RW `clicintie[4095:4064]`  | settings for interrupts 4095 through 4064
+|===
+
+
+== Same Privilege Mode Interrupt Preemption Support - smclic
+
+This extension requires smclicincr extension.
+
+The smclic extension extends interrupt preemption to support up to 256 interrupt
+levels for each privilege mode, where higher-numbered interrupt levels
+can preempt lower-numbered interrupt levels.  Interrupt level 0
+corresponds to regular execution outside of an interrupt handler.
+Levels 1--255 correspond to interrupt handler levels. Platform

```

Comment:
the statement on platform levels is not needed here. otherwise you could put this anywhere. Also, I believe an ISA profile would also be allowed to restrict this.